internal/mailer/mailer_mailgun.go

```diff
@@ -0,0 +1,52 @@
+package mailer
+
+import (
+	"context"
+	"time"
+
+	"github.com/mailgun/mailgun-go/v3"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Mailgun mailer
+type Mailgun struct {
+	client *mailgun.MailgunImpl
+}
+
+// NewMailgun constructs a new Mailgun provider
+func NewMailgun(client *mailgun.MailgunImpl) (Mailer, error) {
+	ml := &Mailgun{
+		client: client,
+	}
+
+	return ml, nil
+}
+
+// SendSuggestion to send a given to the user
+func (m *Mailgun) SendSuggestion(email string, html string) error {
+	logger := logrus.WithFields(logrus.Fields{
+		"logger":     "mailers/Mailgun.SendSuggestion",
+		"user.email": email,
+	})
+
+	logger.Info("sending suggestion to the user")
+
+	msg := m.client.NewMessage(
+		"Kostas Bariotis <newsletter@sandboxbd7bb1a1171e4f41abb06173d239f07d.mailgun.org>",
+		"Your weekly GitHub newsletter",
```

Comment:
Just FYI, these should be part of the config, if not now we should make sure we move them there at some point.